### PR TITLE
HPCC-8603 When assign to SELF.x.y also check if SELF.x assigned

### PR DESCRIPTION
**Pull request by ghalliday**
Signed-off-by: Gavin Halliday gavin.halliday@lexisnexis.com

### CODE REVIEW

**Comment by ghalliday** [comment]
@shamser please review

File: ecl/hql/hqlgram2.cpp [hunk 1]
```diff
@@ -1404,11 +1440,9 @@ void HqlGram::addAssignment(attribute & target, attribute &source)
     }
     else if (targetOp == no_select)
     {
-        // self.* := expr;      assertex(targetExpr->getOperator()==no_select);
-        if (findAssignment(targetExpr))
+        // self.* := expr;
+        if (checkAlreadyAssigned(target, targetExpr))
```
Comment (shamser) [inline ecl/hql/hqlgram2.cpp]: If checkAlreadyAssigned produces a warning message, the assignment will no longer be made.  Is that ok?

File: ecl/regress/issue8603.ecl [hunk 2]
```diff
@@ -0,0 +1,7 @@
+r  := { INTEGER y };
+
+d := DATASET(1, TRANSFORM({ r x },   SELF.x.y := COUNTER));
+
+d2 := DATASET(1, TRANSFORM({ r x }, SELF.x := d[1].x, SELF.x.y := COUNTER));
+
+OUTPUT(d2);
```
Comment (shamser) [inline ecl/regress/issue8603.ecl]: Requires a newline at the end of the last line.

**Comment by shamser** [comment]
@ghalliday Code reviewed.. a couple of comments above.

File: ecl/hql/hqlgram2.cpp (hunk 1 above)
Comment (ghalliday) [inline ecl/hql/hqlgram2.cpp]: Yes that is what is required - otherwise you will get internal errors later on.

**Comment by ghalliday** [comment]
@richardkchapman I think ok to merge, unless you want me to fix the missing end of line?

**Pull request merged**